cloud/passportauth/src/auth/PassportSetup.ts

```diff
@@ -0,0 +1,79 @@
+import * as express from 'express';
+import * as session from 'express-session';
+import {SessionOptions} from 'express-session';
+import * as passport from 'passport';
+import {Strategy} from 'passport-local';
+import {UserSecService} from '../user/UserSec';
+
+export interface Setup {
+  init(app: express.Express, sessionOpts: SessionOptions): void;
+  setStrategy(): void;
+  setSerializeUser(): void;
+  setDeserializeUser(): void;
+}
+
+export class PassportSetup implements Setup {
+  protected userSec: UserSecService;
+
+  constructor(protected readonly userSecService: UserSecService) {
+    this.userSec = userSecService;
+  }
+
+  /**
+   * Initializes passport authentication on an Express application.
+   * This initializes the Express session, passport as well as
+   * setting the local strategy, serializeUser and deserializeUser function.
+   *
+   * @param app {express.Express} An express app used by the application.
+   * @param sessionOpts {SessionOptions} Configuration used by Express session
+   */
+  public init(app: express.Express, sessionOpts: SessionOptions) {
+    app.use(session(sessionOpts));
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    this.setStrategy();
+    this.setSerializeUser();
+    this.setDeserializeUser();
+  }
+
+  /**
+   * Sets the local strategy used by passport authentication.
+   */
+  public setStrategy() {
+    passport.use(new Strategy((loginId, password, done) => {
+      this.userSec.getUserId(loginId).then((userId: string) => {
+        if (!userId) {
+          return done(null, false);
+        } else {
+          this.userSec.comparePassword(userId, password).then((valid: boolean) => {
+            return valid ? done(null, userId) : done(null, false);
```

Comment:
Here is some type inconsistency `done(null, userId) : done(null, false);` one is userId: string and other is false: boolean